### PR TITLE
Add gRPC integration tests

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Addresses CryptoAlg-2163

### Description of changes: 
Add a new integration test that builds gRPC with AWS-LC.

### Call-outs:
This change only runs the TLS/SSL/Cert tests as running all the gRPC tests take over an hour even on a very large instance. 

### Testing:
Tested locally and in the CI. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/cdk/util/iam_policies.py [hunk 1]
```diff
@@ -95,7 +95,15 @@ def code_build_batch_policy_in_json(project_ids):
...
                 "Resource": resources
-            }
+            },{
+                "Effect": "Allow",
+                "Action": [
+                    "ecr:GetDownloadUrlForLayer",
+                    "ecr:GetAuthorizationToken",
+                    "ecr:BatchGetImage"
+                ],
+                "Resource": "*"
```
Comment (WillChilds-Klein) [inline tests/ci/cdk/util/iam_policies.py:105]: do we know the registry ID? `ecr:GetAuthorizationToken` on a resource wildcard is pretty broad.

File: tests/ci/integration/run_grpc_integration.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#  |
+#  - SCRATCH_FOLDER
+#    |
+#    - grpc
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+#    - GRPC_BUILD_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SYS_ROOT}/"GRPC_BUILD_ROOT"
```
[the remark this answers is not in the document]
Comment (justsmth) [reply tests/ci/integration/run_grpc_integration.sh:19]: Should this be [`${SRC_ROOT}`](https://github.com/aws/aws-lc/blob/a8b2a0c87b3a9e0913f18a5cb935f7b69b224850/tests/ci/common_posix_setup.sh#L12)? 

There's in inconsistency in our scripts -- some base `SCRATCH_FOLDER` on [`SYS_ROOT`](https://github.com/aws/aws-lc/blob/341a240a35ea405687aee0641567a236c97848e4/tests/ci/integration/run_s2n_integration.sh#L21) and some of [`SRC_ROOT`](https://github.com/aws/aws-lc/blob/4cb82544709ef22fb7608cb93836b7a2051729aa/tests/ci/integration/run_postgres_integration.sh#L17).

File: tests/ci/integration/run_grpc_integration.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+mkdir -p ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER}
+
+git clone --depth 1 https://github.com/grpc/grpc.git ${GRPC_SRC_FOLDER}
+cd ${GRPC_SRC_FOLDER}
+git submodule update --init
```
Comment (justsmth) [inline tests/ci/integration/run_grpc_integration.sh:32]: This should recurse:
```
git submodule update --recursive --init
```

File: tests/ci/integration/run_grpc_integration.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewhop) [reply tests/ci/integration/run_grpc_integration.sh:19]: I'm leaning towards keeping the integration builds outside aws-lc and keeping all the artifacts grouped together.

File: tests/ci/integration/run_grpc_integration.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+aws_lc_build ${SRC_ROOT} ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} -DBUILD_SHARED_LIBS=1
```
[the remark this answers is not in the document]
Comment (andrewhop) [reply tests/ci/integration/run_grpc_integration.sh:34]: Yes

**Review by WillChilds-Klein — DISMISSED** [review]

File: tests/ci/integration/run_grpc_integration.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+mkdir -p "${GRPC_SRC_FOLDER}/cmake/build"
+cd "${GRPC_SRC_FOLDER}/cmake/build"
+cmake -GNinja -DgRPC_BUILD_TESTS=ON -DCMAKE_BUILD_TYPE=Release  -DgRPC_SSL_PROVIDER=package  -DBUILD_SHARED_LIBS=ON  -DOPENSSL_ROOT_DIR="${AWS_LC_INSTALL_FOLDER}" ../..
+ninja
+
+# grpc tests expect to use relative paths to certificates and test files
+cd "${GRPC_SRC_FOLDER}"
+python3 tools/run_tests/start_port_server.py
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_grpc_integration.sh:43]: does this always start up immediately? if there's a potential for a race we could sit in a timeout-bound loop doing `nc -vz 127.0.0.1 443` (or whatever port it is) until the server is up. might not be worth the additional complexity unless we observe this to be a problem, though

**Review by WillChilds-Klein — APPROVED** [review]

File: tests/ci/integration/run_grpc_integration.sh (hunk 5 above)
Comment (andrewhop) [reply tests/ci/integration/run_grpc_integration.sh:43]: Yeah, it appears the python script waits to exit until the server is up.

**Review by samuel40791765 — DISMISSED** [review]

**Review by samuel40791765 — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**